backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -34,7 +38,7 @@
     return engagementRepository.findByStaffId(staffId).stream()
         .filter(engagement ->
             (engagement.getStart().equals(start)
-                || engagement.getStart().isAfter(startTime))
+                || !engagement.getStart().isBefore(startTime))
           && (engagement.getEnd().equals(endTime)
                 || engagement.getEnd().isBefore(endTime)))
```

Comment:
```
&& !engagement.getEnd().isAfter(endTime)
```

Reply:
Ah yep I see good point, I'll update this